samples/order/order.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"gocloud.dev/blob"
+	"gocloud.dev/docstore"
+	"gocloud.dev/pubsub"
+)
+
+var (
+	requestTopicURL  = flag.String("request-topic", "mem://requests", "gocloud.dev/pubsub URL for request topic")
+	requestSubURL    = flag.String("request-sub", "mem://requests", "gocloud.dev/pubsub URL for request subscription")
+	responseTopicURL = flag.String("response-topic", "mem://responses", "gocloud.dev/pubsub URL for response topic")
+	responseSubURL   = flag.String("response-sub", "mem://responses", "gocloud.dev/pubsub URL for response subscription")
+	bucketURL        = flag.String("bucket", "", "gocloud.dev/blob URL for image bucket")
+	collectionURL    = flag.String("collection", "mem://orders/ID", "gocloud.dev/docstore URL for order collection")
+	// TODO(jba): uncomment after adding frontend
+	// runProcessor     = flag.Bool("processor", true, "run the image processor")
+)
+
+func main() {
+
+	// TODO(jba): add frontend
+	flag.Parse()
+	_, processor, cleanup, err := setup()
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer cleanup()
+	if err := processor.run(context.Background()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func setup() (_ *frontend, _ *processor, cleanup func(), err error) {
+	var cleanups []func()
+	defer func() {
```

Comment:
This looks OK, but is probably too fancy for a sample. Maybe add a TODO to simplify.

Reply:
done